src/crypto/crypto.h

```diff
@@ -335,8 +335,16 @@ namespace crypto {
 
   inline bool operator<(const public_key &p1, const public_key &p2) { return memcmp(&p1, &p2, sizeof(public_key)) < 0; }
   inline bool operator>(const public_key &p1, const public_key &p2) { return p2 < p1; }
+  inline bool operator<(const key_image &p1, const key_image &p2) { return memcmp(&p1, &p2, sizeof(key_image)) < 0; }
+  inline bool operator>(const key_image &p1, const key_image &p2) { return p2 < p1; }
 }
 
+// type conversions for easier calls to sc_add(), sc_sub(), hash functions
+inline unsigned char* to_bytes(crypto::ec_scalar &scalar) { return &reinterpret_cast<unsigned char&>(scalar); }
```

Comment:
These should be dropped in favor of `epee::to_byte_span()` usage. Or just redirect to that function and return a `span` here, although I don't see a purpose of re-creating those existing functions.

Reply:
If I use the span stuff directly, then:
```
sc_add(to_bytes(a), to_bytes(b), to_bytes(c));  //old
sc_add(epee::as_mut_byte_span(a).data(), epee::as_byte_span(b).data(), epee::as_byte_span(c).data());  //new
```

Using a `span` internally means you have to construct a `span` every time `to_bytes` is called, which is not ideal for one of the main use cases - high performance crypto algorithms.

Reply:
The compiler should optimize away all references to the span if only the pointer is used. This is pretty basic stuff for the compiler actually.

I wasn't aware of the context though - these existing functions take a pointer with no other context anyway. My frustration with adding these functions is the the temptation to use these in non-crypto related functions. I guess reviewers will have to catch this.

Reply:
It's hard for me to imagine a case where `to_bytes()` would be used for anything other than exposing a raw pointer to C-style crypto functions.

Reply:
I think I fall on the side of @UkoeHB here. Most of the underlying C crypto functions already use "unsafe" pointers insofar as the size of the area passed is implied by the contract of the function, not enforced at compile-time or runtime. Unless we want to rewrite these functions, using `epee::span` would just add an unused wrapper on top of a pointer which may or may not get optimized out by the compiler.